### PR TITLE
Don't strip pointed-to type information in IL stubs

### PR DESCRIPTION
**Pull request by jkoritzinsky**
With the new Swift interop support, we need the JIT to be able to see the `SwiftError*` types in the unmanaged call signatures, even in the P/Invoke stub case (as the JIT always generates P/Invoke stubs in debug code).

This PR updates the handling for pointer and byref types (as the existing Swift tests use `ref SwiftError` in some cases) so that the type in the target signature will be `T*` instead of `nint` for a managed `T*` or `ref T` argument.

This PR also adjusts some of the IL generation to use `conv.u` instead of `conv.i` to match Roslyn's IL codegen for similar manually-written code.

### CODE REVIEW

File: src/coreclr/vm/stubgen.cpp [hunk 1]
```diff
@@ -2639,22 +2639,16 @@ void ILStubLinker::TransformArgForJIT(LocalDesc *pLoc)
 
         case ELEMENT_TYPE_PTR:
         {
-#ifdef TARGET_X86
-            if (pLoc->bIsCopyConstructed)
```
Comment (AaronRobinsonMSFT) [inline src/coreclr/vm/stubgen.cpp:2643]: With this, I think you can remove the field entirely.

**Review by AaronRobinsonMSFT — COMMENTED** [review]
Can you confirm we have a test that has a range of `ELEMENT_TYPE_*` that are blittable?

We really should have a selection of types to ensure we're covered.

```csharp
int*
float*
decimal*
nint*
<VALUE_TYPE>*
<VALUE_TYPE_WITH_GENERIC>*
<GENERIC_PARAM_CONSTRAINED_TO_UNMANAGED>*
```

**Comment by jkoritzinsky** [comment]
I've added some additional tests. I can't add a `T*` case as we currently don't have a scenario that supports a generic type argument parameter here (the generic calli PR will enable that case).

**Review by AaronRobinsonMSFT — APPROVED** [review]
LGTM if everything is green. Thanks for adding the scenarios!

**Pull request merged**